aten/src/ATen/native/cpu/MultinomialKernel.cpp

```diff
@@ -94,27 +90,12 @@ void multinomial_apply(Tensor& result, const Tensor& self, const int64_t n_sampl
 
       /* Once a sample is drawn, it cannot be drawn again. ie sample without replacement */
       if (!with_replacement && j < n_sample - 1) {
-        /* update cumulative distribution so that sample cannot be drawn again */
-        scalar_t diff;
-        scalar_t new_val = 0;
-        scalar_t sum;
-
-        if (sample_idx != 0) {
-          new_val = cum_dist_ptr[(sample_idx - 1) * cum_dist_stride_0];
-        }
-        /* marginal cumulative mass (i.e. original probability) of sample */
-        diff = cum_dist_ptr[sample_idx * cum_dist_stride_0] - new_val;
-        /* new sum of marginals is not one anymore... */
-        sum = 1.0 - diff;
-        for (int64_t k = 0; k < n_categories; k++) {
-          new_val = cum_dist_ptr[k * cum_dist_stride_0];
-          if (k >= sample_idx) {
-            /* remove sampled probability mass from later cumulative probabilities */
-            new_val -= diff;
-          }
-          /* make total marginals sum to one */
-          new_val /= sum;
-          cum_dist_ptr[k * cum_dist_stride_0] = new_val;
+        self_copy_ptr[sample_idx] = 0;
+        sum = 0;
+        for (int64_t j = 0; j < n_categories; j++) {
+          val = self_copy_ptr[j];
+          sum += val;
+          cum_dist_ptr[j * cum_dist_stride_0] = sum;
```

Comment:
Maybe the variables
```
auto cum_dist_stride_0 = cum_dist.stride(0);
auto self_copy_stride_0 = self_copy.stride(0);
``` 
should be be removed? I think there are no cases in which newly created tensors could be non-contiguous, are there? At least in ln 93 `self_copy_ptr` is indexed without it, so this should also work for the 1d `cum_dist` tensor.

Reply:
Agreed

Reply:
If we really want to be certain, we can construct the tensor and then assert it is contiguous right after the construction.